pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -862,105 +862,137 @@ func TestPriorityQueue_Pop(t *testing.T) {
-		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPodInfo.Pod.Name, p.Pod.Name)
-	}
 
-	// Updating a pod that is in backoff queue and it is still backing off
-	// pod will not be moved to active queue, and it will be updated in backoff queue
-	podInfo := q.newQueuedPodInfo(medPriorityPodInfo.Pod)
-	if err := q.podBackoffQ.Add(podInfo); err != nil {
-		t.Errorf("adding pod to backoff queue error: %v", err)
-	}
-	q.Update(logger, podInfo.Pod, podInfo.Pod)
-	rawPodInfo, err := q.podBackoffQ.Pop()
-	podGotFromBackoffQ := rawPodInfo.(*framework.QueuedPodInfo).Pod
-	if err != nil || podGotFromBackoffQ != medPriorityPodInfo.Pod {
-		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPodInfo.Pod.Name, podGotFromBackoffQ.Name)
+	tests := []struct {
+		name      string
+		expectedQ string
+		// addedToNominatedPods is whether a Pod from the test case should be registered as a nominated Pod in the nominator.
+		addedToNominatedPods bool
+		// prepareFunc is the function called to prepare pods in the queue before the test case calls Update().
+		// This function returns three values;
+		// - oldPod/newPod: each test will call Update() with these oldPod and newPod.
+		// - wantUpdatedPod: an expected QueuedPodInfo in expectedQ after Update()
+		prepareFunc func(logger klog.Logger, q *PriorityQueue) (oldPod, newPod *v1.Pod, wantUpdatedPod *framework.QueuedPodInfo)
+	}{
+		{
+			name:      "add highPriorityPodInfo to activeQ",
+			expectedQ: activeQ,
+			prepareFunc: func(logger klog.Logger, q *PriorityQueue) (oldPod, newPod *v1.Pod, wantUpdatedPod *framework.QueuedPodInfo) {
+				return nil, highPriorityPodInfo.Pod, q.newQueuedPodInfo(highPriorityPodInfo.Pod)
+			},
+		},
+		{
+			name:      "Update pod that oldPod doesn't exist in the queue",
+			expectedQ: activeQ,
+			prepareFunc: func(logger klog.Logger, q *PriorityQueue) (oldPod, newPod *v1.Pod, wantUpdatedPod *framework.QueuedPodInfo) {
+				updatedPod := medPriorityPodInfo.Pod.DeepCopy()
+				updatedPod.Annotations["foo"] = "test"
+				return medPriorityPodInfo.Pod, updatedPod, q.newQueuedPodInfo(updatedPod)
+			},
+		},
+		{
+			name:                 "Update highPriorityPodInfo and add a nominatedNodeName to it",
+			expectedQ:            activeQ,
+			addedToNominatedPods: true,
+			prepareFunc: func(logger klog.Logger, q *PriorityQueue) (oldPod, newPod *v1.Pod, wantUpdatedPod *framework.QueuedPodInfo) {
+				return highPriorityPodInfo.Pod, highPriNominatedPodInfo.Pod, q.newQueuedPodInfo(highPriNominatedPodInfo.Pod)
+			},
+		},
+		{
+			name:      "When updating a pod that is already in activeQ, the pod should remain in activeQ after Update()",
+			expectedQ: activeQ,
+			prepareFunc: func(logger klog.Logger, q *PriorityQueue) (oldPod, newPod *v1.Pod, wantUpdatedPod *framework.QueuedPodInfo) {
+				err := q.Update(logger, nil, highPriorityPodInfo.Pod)
+				if err != nil {
+					t.Errorf("add pod %s error: %v", highPriorityPodInfo.Pod.Name, err)
```

Comment:
either pass `t` as a parameter or return the error and handle it in the test code. Otherwise you will be failing all cases, instead of just one.